python_modules/libraries/dagster-tableau/dagster_tableau/resources.py

```diff
@@ -0,0 +1,154 @@
+from abc import abstractmethod
+from typing import Any, Dict, Optional
+
+import requests
+from dagster import ConfigurableResource, InitResourceContext
+from dagster._utils.cached_method import cached_method
+from pydantic import Field, PrivateAttr
+
+TABLEAU_API_VERSION = "3.23"
+
+
+class BaseTableauWorkspace(ConfigurableResource):
+    """Base class to represent a workspace in Tableau and provides utilities
+    to interact with the Tableau API.
+    """
+
+    personal_access_token_name: str = Field(
+        ..., description="The name of the personal access token used to connect to Tableau API."
+    )
+    personal_access_token_value: str = Field(
+        ..., description="The value of the personal access token used to connect to Tableau API."
+    )
+    site_name: str = Field(..., description="The name of the Tableau site to use.")
+
+    _api_token: Optional[str] = PrivateAttr(default=None)
+    _site_id: Optional[str] = PrivateAttr(default=None)
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
```

Comment:
@benpankow @yuhan correct me if I'm wrong here, but I think our typical pattern for resources is that we require a method invocation to connect, rather than doing it automatically at step startup time?  E.g. `SnowflakeResource` exposes a `get_connection` method instead of automatically connecting at step startup.